cdap-docs/build.sh

```diff
@@ -20,283 +20,210 @@
-    if [ "${javadocs}" == "${WITH}" ]; then
-      run_command javadocs
-    fi
-    if [ "${cli_docs}" == "${WITH}" ]; then
-      run_command docs-cli
-    fi
-  else
-    check_build_rst
-  fi
-  if [ "${doc_type}" == "${GITHUB}" -o "${doc_type}" == "${GITHUB_ONLY}" ]; then
-    run_command docs-github-part ${source_path}
-  elif [ "${doc_type}" == "${WEB}" -o "${doc_type}" == "${WEB_ONLY}" ]; then
-    run_command docs-web-part ${source_path}
-  fi
-  set_display_version
-  display_messages_file
+}
+
+function build_docs_set() {
+  local warnings
+  local title="Building Docs Set"
+  display_start_title ${title}
+  
+  clean_targets
+  clear_messages_file
+  build_docs_first_pass
+  clear_messages_file
+  build_javadocs docs
+  build_docs_cli
+  build_docs_second_pass
+  build_docs_package
+  
+  set_and_display_version
+  display_messages
   warnings=$?
   cleanup_messages_file
-  echo "--------------------------------------------------------"
-  ring_bell "Completed build of \"${doc_type}\""
-  echo "========================================================"
-  echo "========================================================"
-  echo
-  return ${warnings}
+  display_end_title_bell ${title}
+  exit ${warnings}
+}
+
+function build_docs_only() {
+  local warnings
+  local title="Building Docs Only"
+  display_start_title ${title}
+  
+  clear_messages_file
+  build_docs_second_pass
+  
+  set_and_display_version
+  display_messages
+  warnings=$?
+  cleanup_messages_file
+  display_end_title_bell ${title}
+  exit ${warnings}
+}
+
+function build_docs_first_pass() {
+  local title="Building Docs First Pass"
+  display_start_title ${title}
+
+  build_docs_inner_level build-docs
+  
+  display_end_title ${title}
+}
+
+function build_docs_second_pass() {
+  local title="Building Docs Second Pass"
+  display_start_title ${title}
+
+  build_docs_inner_level build-web
+  build_docs_outer_level ${GOOGLE_TAG_MANAGER_CODE}
+  copy_docs_inner_level
+
+  display_end_title ${title}
 }
 
 function build_javadocs() {
   local javadoc_type=${1}
-  echo "========================================================"
-  echo "Building Javadocs: '${javadoc_type}'"
-  echo "--------------------------------------------------------"
-  echo
-  if [ ${NO_JAVADOCS} ]; then
+  local title="Building Javadocs: '${javadoc_type}'"
+  display_start_title ${title}
+
+  local warnings
+  if [[ -n ${NO_JAVADOCS} ]]; then
     echo_red_bold "Javadocs disabled."
   else
-    build_javadocs_api ${javadoc_type}
     USING_JAVADOCS="true"
     export USING_JAVADOCS
+    if [ "${javadoc_type}" == "${ALL}" ]; then
+      local javadoc_run="mvn javadoc:aggregate -P release"
+    else
+      local javadoc_run="mvn clean site -P templates"
+    fi
+    if [ "${DEBUG}" == "${TRUE}" ]; then
+      local debug_flag="-X"
+    else
+      local debug_flag=''
+    fi
+    set_mvn_environment
+    echo "JAVA_HOME: ${JAVA_HOME}"
+    echo "JAVA Version:"
+    java -version
```

Comment:
Only if the `java` in the PATH is the same one in `JAVA_HOME`... maybe set a `JAVA` variable with the resolved PATH to the correct `java` binary, like [CDAP does](https://github.com/caskdata/cdap/blob/develop/cdap-common/bin/functions.sh#L286-L310)... or, you could actually source the `functions.sh` file and use the function, directly.